editor/src/components/canvas/controls/select-mode/flex-gap-control.tsx

```diff
@@ -0,0 +1,285 @@
+          )
+        })}
+      </div>
+    </CanvasOffsetWrapper>
+  )
+})
+
+interface GapControlSizeConstants {
+  dragBorderWidth: number
+  paddingIndicatorOffset: number
+  hitAreaPadding: number
+  borderWidth: number
+}
+
+const DefaultGapControlSizeConstants: GapControlSizeConstants = {
+  dragBorderWidth: 1,
+  borderWidth: 0.5,
+  paddingIndicatorOffset: 10,
+  hitAreaPadding: 5,
+}
+
+const gapControlSizeConstants = (
+  constants: GapControlSizeConstants,
+  scale: number,
+): GapControlSizeConstants => ({
+  dragBorderWidth: constants.dragBorderWidth / scale,
+  borderWidth: constants.borderWidth / scale,
+  paddingIndicatorOffset: constants.paddingIndicatorOffset / scale,
+  hitAreaPadding: constants.hitAreaPadding / scale,
+})
+
+interface GapControlSegmentProps {
+  hoverStart: React.MouseEventHandler
+  hoverEnd: React.MouseEventHandler
+  handleHoverStart: (_: string) => void
+  handleHoverEnd: () => void
+  onMouseDown: React.MouseEventHandler
+  bounds: CanvasRectangle
+  flexDirection: SimpleFlexDirection
+  gapValue: CSSNumber
+  indicatorShown: string | null
+  path: string
+  indicatorColor: string
+  scale: number
+  isDragging: boolean
+  backgroundShown: boolean
+}
+
+const GapControlSegment = React.memo<GapControlSegmentProps>((props) => {
+  const {
+    hoverStart,
+    hoverEnd,
+    handleHoverEnd,
+    handleHoverStart,
+    onMouseDown,
+    indicatorShown,
+    bounds,
+    isDragging,
+    gapValue,
+    flexDirection,
+    indicatorColor,
+    scale,
+    path,
+    backgroundShown,
+  } = props
+
+  const { dragBorderWidth, hitAreaPadding, paddingIndicatorOffset, borderWidth } =
+    gapControlSizeConstants(DefaultGapControlSizeConstants, scale)
+  const { width, height } = handleDimensions(flexDirection, scale)
+
+  const handleHoverStartI = React.useCallback(
+    () => handleHoverStart(path),
+    [handleHoverStart, path],
+  )
+
+  const shouldShowIndicator = React.useCallback(
+    (p: string) => !isDragging && indicatorShown === p,
+    [indicatorShown, isDragging],
+  )
+
+  const shouldShowBackground = !isDragging && backgroundShown
+
+  return (
+    <div
+      key={path}
+      onMouseEnter={hoverStart}
+      onMouseLeave={hoverEnd}
+      style={{
+        position: 'absolute',
+        left: bounds.x,
+        top: bounds.y,
+        width: bounds.width,
+        height: bounds.height,
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        border: isDragging ? `${dragBorderWidth}px solid ${indicatorColor}` : undefined,
+        ...(shouldShowBackground ? StripedBackgroundCSS(indicatorColor, scale) : {}),
+      }}
+    >
+      <div
+        data-testid={FlexGapControlHandleTestId}
+        style={{ padding: hitAreaPadding, cursor: cursorFromFlexDirection(flexDirection) }}
+        onMouseDown={onMouseDown}
+        onMouseEnter={handleHoverStartI}
+        onMouseLeave={handleHoverEnd}
+      >
+        {shouldShowIndicator(path) && (
```

Comment:
I believe this should be using the `when` utility function, we try to avoid exploiting truthiness especially with `&&` and `||`.

Reply:
great suggestion, done!